functions/src/triggers/stages.triggers.ts

```diff
@@ -149,17 +149,47 @@ export const publishStageData = onDocumentWritten(
           .get();
         const publicData = publicDoc.data() as VoteForLeaderStagePublicData;
 
-        // Update participant rankings
         const participantRankings = publicData.participantRankings;
         participantRankings[participantPublicId] = data.rankings;
 
-        // TODO: When determining current leader,
-        // only use participants whose scores are in the top two
-        // for willingness to lead (WTL).
-        //
-        // Determine top two scores from WTL, then filter participantRankings
-        // by participants who have those scores.
-        const currentLeader = getCondorcetElectionWinner(participantRankings);
+        let currentLeader = getCondorcetElectionWinner(participantRankings);
```

Comment:
Instead of calling this early, could we only call getConcorcetElectionWinner once determining which participants should be sent in? E.g., updating participantRankings if needed and then call at the end

Reply:
Done!